<launch>
  <arg name="dataset_name" value="oakd"/>

  <arg name="online" default="true" />
  <arg name="log_output" default="true"/>
  <arg name="verbosity" default="1"/>
  <arg name="debug" default="false"/>
  <arg name="should_use_sim_time" default="false"/>

  <arg name="camera_ns" default="/stereo_inertial_publisher"/>
  <arg name="use_lcd" default="false"/>

  <arg name="log_path" value="$(find kimera_vio_ros)/output_logs/$(arg dataset_name)"/>

  <include file="$(find kimera_vio_ros)/launch/kimera_vio_ros.launch" pass_all_args="true">
      <!-- Options -->
      <arg name="include_pipeline_flags" value="true"/>
      <arg name="viz_type" value="1"/>
      <arg name="use_rviz" value="false"/>
      <!-- Other settings -->
      <arg name="log_gt_data" value="false"/>
      <arg if="$(arg log_output)" name="log_output_path" value="$(arg log_path)"/>
      <!-- Subscribed Topics -->
      <arg name="left_cam_topic" value="$(arg camera_ns)/left/image_rect"/>
      <arg name="depth_cam_topic" value="$(arg camera_ns)/stereo/depth"/>
      <arg name="imu_topic"  value="$(arg camera_ns)/imu"/>
      <!-- Only used when parsing a rosbag -->
      <arg name="odom_frame_id"     value="world"/>
      <arg name="base_link_frame_id" value="oak-d_frame"/>
      <arg name="map_frame_id"       value="map"/>
      <arg name="left_cam_frame_id"  value="oak_left_camera_frame"/>
      <arg name="right_cam_frame_id" value="oak_right_camera_frame"/>
  </include>
  
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find kimera_vio_ros)/rviz/kimera_vio_oakd.rviz" />
</launch>
